pkg/platform/github.go

```diff
@@ -150,6 +154,96 @@ func (g *GitHub) AssignReviewers(ctx context.Context, input *AssignReviewersInpu
 	return &result, nil
 }
 
+type pullRequestReview struct {
+	Author struct {
+		Login string
+	}
+	State string
+}
+
+type latestApproverQuery struct {
+	Repository struct {
+		PullRequest struct {
+			LatestReviews struct {
+				Nodes []pullRequestReview
+			} `graphql:"latestReviews(first: 100)"`
+		} `graphql:"pullRequest(number: $number)"`
+	} `graphql:"repository(owner: $owner, name: $repo)"`
+}
+
+type member struct {
+	Login string
+}
+type team struct {
+	Name    string
+	Members struct {
+		Nodes []member
+	} `graphql:"members(first: 100)"`
+}
+type organizationTeamsAndMembershipsQuery struct {
+	Organization struct {
+		Teams struct {
+			Nodes []team
+		} `graphql:"teams(first: 100)"`
+	} `graphql:"organization(login: $owner)"`
+}
+
+// GetLatestApprovers retrieves the users whose latest review for a pull request
+// is an approval. It also returns the teams and subteams that the user
+// approvers are members of to indicate approval on behalf of those teams. Note,
+// if a reviewer leaves a comment OR requests changes after a previous approval,
+// then the reviewer will not be included in the "latest" approvers list.
```

Comment:
Is this the behavior we want? I know we've touched on this, but I would think any approval is an approval.

Reply:
I just tested the behavior. It looks like a comment will not change the status of the review, so a prior approval will still be in effect. If the approval is dismissed or followed by a requested changes, then the approval won't be counted.

That does seem like the behavior we want.

Reply:
+1, thanks!